File: Практика - Timus Online Judge/1247/1247.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    internal class Program
    {
        static void Main(string[] args)
        {
            //int s, n;
            //s = Convert.ToInt32(Console.ReadLine());
            //n = Convert.ToInt32(Console.ReadLine());
            int[] A = Array.ConvertAll(Console.ReadLine().Split(), int.Parse); // A[0] - s; a[1] - n

            int[] data = new int[A[0]];

            for (int i = 0; i < A[0]; i++)
            {
                data[i] = Convert.ToInt32(Console.ReadLine());
            }

            int sum = 0;
            for (int i = 0; i < A[0]; i++)
            {
                sum += data[i];
                if (sum < (i + 1))
                {
                    Console.WriteLine("NO");
                    return;
                }
            }

            sum = 0;
            for (int i = 0; i < A[0]; i++)
            {
                sum += data[A[0] - 1 - i];
                if (sum < (i + 1))
                {
                    Console.WriteLine("NO");
                    return;
                }
            }

            Console.WriteLine("YES");

            //string input = Console.ReadLine();
        }
    }
}

File: Практика - Timus Online Judge/1247/1247.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std; 

// Проверяем является ли заданная последовательность чисел корректной. Суммируем элементы по порядку в обе стороны и
// и проверяем, сумма < i+1 - если условие нарушено хотя бы для одного элемента -> NO

int main()
{
    int s, n;
    cin >> s >> n;

    vector<int> data(s);

    for (int i = 0; i < s; i++) // записываем элементы последовательности в вектор
        cin >> data[i];

    int sum = 0;
    for (int i = 0; i < s; i++) // цикл для вычисления суммы элементов вектора
    {
        sum += data[i];
        if (sum < (i + 1)) // если сумма на i-м шаге меньше, чем i + 1 -> выводим "NO"
        {
            cout << "NO" << endl;
            return 0;
        }
    }

    sum = 0; // обнуляем переменную sum
    for (int i = 0; i < s; i++) // цикл для вычисления суммы элементов вектора в обратном порядке
    {
        sum += data[s - 1 - i];
        if (sum < (i + 1))
        {
            cout << "NO" << endl;
            return 0;
        }
    }

    cout << "YES" << endl; // проверки пройдены успешно

    return 0;
}
